src/components/Visualization/Visualization.js

```diff
@@ -79,19 +79,37 @@ export const Visualization = ({
         }
     )
 
-    const isInModal = !!filters?.relativePeriodDate
+    const visualizationRef = useRef(visualization)
+
+    const setPage = useCallback(
```

Comment:
Is this `useCallback` needed?
There isn't any logic using those dependencies, they are just passed to the `setSorting`, so in practice I don't think there is any advantage in using `useCallback`.

Reply:
Or is that needed because `setPage` is a dependency in a `useEffect`? 🤔

Reply:
@edoardo Yes, otherwise there's an endless loop.